File: RefinerySystem/Refinery/src/models/objects/Route.ts
import { FuelType } from "./FuelType";
import { Truck } from "./Truck";

export interface Route {
    id?: number;
    fuel_type?: FuelType;
    fuel_type_id?: number;
    truck_id?: number;
    truck?: Truck;
    name: string;
    fuel_quantity: number;
    fuel_price: number;
    date?: string;
    completed?: boolean;
}

File: RefinerySystem/RefineryAPI/refinery/models/route.py
from django.db import models


class Route(models.Model):
    name = models.CharField(max_length=100)
    fuel_quantity = models.FloatField()
    fuel_price = models.FloatField()
    fuel_type = models.ForeignKey('FuelType', on_delete=models.CASCADE)
    truck = models.ForeignKey('Truck', on_delete=models.CASCADE)
    date = models.DateField(auto_now_add=True)
    completed = models.BooleanField(default=False)
